### PR TITLE
CI: Tweak workflow to make it simpler to sync with other repos

### PR DESCRIPTION
**Pull request by yamt**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -16,32 +16,6 @@ on:
   pull_request:
 
 jobs:
-  check:
```
Comment (xiaoxiang781216) [inline .github/workflows/main.yml]: Should we change line 13 to:
name: PR Check CI (build)
and rename main.yml to build.yml?

**Comment by liuguo09** [comment]
LGTM. Besides xiang's comment, one thing I worry a little bit is: would we keep on matrix build jobs donot depends on check job always as in https://github.com/apache/incubator-nuttx/pull/552 does? If the answer is yes, it's good and reasonable to split into check.yml and build.yml.

**Comment by yamt** [comment]
@liuguo09 we can merge it back if/when we want to put back the dependency. i separated it for now because at this point it seems little point to run it on testing repo.

File: .github/workflows/main.yml (hunk 1 above)
Comment (yamt) [reply .github/workflows/main.yml]: replied in https://github.com/apache/incubator-nuttx/pull/666

**Comment by liuguo09** [comment]
@yamt I understand. Thanks.

**Pull request merged**